### PR TITLE
fix: Options fields operator comparison.

### PR DESCRIPTION
**Pull request by EdwinBetanc0urt**
## Bug report / Feature

Comparison operators are not displayed in the advanced search.
Disabled the `Preference` and `Change Logs` field options in advanced search

#### Steps to reproduce

1. Open any Window.
2. Show table of records.
3. Open fields of advanced search.
4. Displayed options of fields.

#### Screenshot or Gif
Before this PR:
options-fields-error

After PR:
options-fields

#### Expected behavior
It is expected to be displayed in the field options of the advanced search the comparison operators.

#### Other relevant information
- Your OS: Linux Mint 19.1 Cinnamon x64.
- Web Browser: Mozilla Firefox 85.0.
- Node.js version: 12.20.0.
- adempiere-vue version: 4.3.1.

#### Additional context
Separates the list of field options in a different component (`components/ADempiere/Field/OptionsInField`), written with vue composition api. So that the main field component is cleaner.

Also rename the folder `components/ADempiere/Field/contextMenuField` to `components/ADempiere/Field/OptionsInField`, because it could be confused with contextInfo which is one of the options associated to the field.

### CODE REVIEW

**Comment by oscarinak** [comment]
The pull verification was carried out in the following service "https://deploy-preview-867--stoic-lamport-68decf.netlify.app/" where effectively in the advanced search option there are search operators, however found the following observations:

Verificación_pullR867

-When consulting the options of the operators in the advanced search fields, they are slow.

-When selecting the "Includes" operator and adding a value in the field, this does not perform any search. Example: that operator was selected and added the letter "A" by giving enter and hoping as a result to obtain records that contain that value, and ADempiere does not perform no search

-When the operators window appears and try to return to the advanced search window, the window should disappear and it costs to disappear

Verificación_ParteII_pullR867_

**Comment by oscarinak** [comment]
Bad

**Comment by EdwinBetanc0urt** [comment]
Thank you @oscarinak , I will review and apply the changes from your observations.

File: src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      //     : value,
+      //   isAdvancedQuery: true,
+      //   isSendToServer: !(value === 'NotSend'),
+      //   isSendCallout: false
+      // }
+
+      // root.$store.dispatch('notifyFieldChange', {
+      //   ...sendParameters,
+      //   isChangedOldValue: props.fieldAttributes.componentPath === 'FieldYesNo' && Boolean(value === 'NotSend')
+      // })
```
Comment (elsiosanchez) [inline src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue]: Hi @EdwinBetanc0urt  Please remove this if is unnecessary

**Comment by EdwinBetanc0urt** [comment]
@oscarinak The necessary changes were made in the client to perform the search with the parameters and operators, however with the 3 different cases the result should arrive empty because in this test there is only one record in the table whose name and code is `Test`.


advanced-search


1. Operator NOT_EQUAL, `Name != 'Test'`.
```json
{
  "table_name": "M_Product_Group",
  "filters": [{
    "column_name": "Name",
    "value": "Test",
    "operator": "NOT_EQUAL",
    "values": []
  }],
  "where_clause": "",
  "language": "es"
}
```

1. Operator NOT_LIKE, `Name NOT LIKE 'Test'`.
```json
{
  "table_name": "M_Product_Group",
  "filters": [{
    "column_name": "Name",
    "value": "Test",
    "operator": "NOT_LIKE",
    "values": []
  }],
  "where_clause": "",
  "language": "es"
}
```

1. Operator NULL, `Name IS NULL`.
```json
{
  "table_name": "M_Product_Group",
  "filters": [{
    "column_name": "Name",
    "value": "Test",
    "operator": "NULL",
    "values": []
  }],
  "where_clause": "",
  "language": "es"
}
```

@yamelsenih Requires changes in the backend server to take the filters sent in the advanced search request.

**Review by elsiosanchez — APPROVED** [review]

File: src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue (hunk 1 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue]: @elsiosanchez  This has been removed

File: src/components/ADempiere/Field/OptionsInField/index.vue [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+            </el-button>
+          </el-popover>
+        </el-menu-item>
+      </el-submenu>
+    </el-menu>
+    <span v-else>
+      {{ metadata.name }}
+    </span>
+
+    <!--
```
Comment (yamelsenih) [inline src/components/ADempiere/Field/OptionsInField/index.vue]: Please remove this fragment

File: src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue [hunk 3]
```diff
@@ -0,0 +1,134 @@
+<!--
```
Comment (yamelsenih) [inline src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue]: Please rename this to **FieldOptions**

**Review by yamelsenih — CHANGES_REQUESTED** [review]

**Comment by oscarinak** [comment]
The PR verification cannot be performed because you fail with the service, a gif of the problem is attached to access the service

Verificación_ParteIV_PullR867_Pull

File: src/components/ADempiere/Field/OptionsInField/index.vue (hunk 2 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/Field/OptionsInField/index.vue]: Remove this unused code

File: src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue (hunk 3 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/Field/OptionsInField/operatorComparison/index.vue]: Rename from OptionsInFiend to FieldOptions

**Comment by EdwinBetanc0urt** [comment]
@oscarinak @yamelsenih the required changes were applied

**Comment by oscarinak** [comment]
The verification was carried out under the "Product" window, which has several records, in order to use the operators and this shows information based on the value assigned to the operator, but it still has not performed the search

Verificación_ParteV_Pull_867

**Comment by oscarinak** [comment]
The PR verification was carried out this time it was done in a browser with normal view and in an incognito mode where when using the operators again it does not perform any search, attached gif of the tests you perform

Verificación_ParteVI_Pull_867


incognito mode

Verificación_ParteVII_Pull_867

File: src/components/ADempiere/Field/FieldOptions/calculator/buttons.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // ADempiere-Vue (Frontend) for ADempiere ERP & CRM Smart Business Solution
 // Copyright (C) 2017-Present E.R.P. Consultores y Asociados, C.A.
-// Contributor(s): Edwin Betancourt edwinBetanc0urt@hotmail.com www.erpya.com
```
Comment (yamelsenih) [inline src/components/ADempiere/Field/FieldOptions/calculator/buttons.js]: Please remove this file and just change when touch it

File: src/components/ADempiere/Field/FieldOptions/calculator/index.vue [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <!--
  ADempiere-Vue (Frontend) for ADempiere ERP & CRM Smart Business Solution
  Copyright (C) 2017-Present E.R.P. Consultores y Asociados, C.A.
- Contributor(s): Edwin Betancourt edwinBetanc0urt@hotmail.com www.erpya.com
+ Contributor(s): Edwin Betancourt EdwinBetanc0urt@oulook.com www.erpya.com
```
Comment (yamelsenih) [inline src/components/ADempiere/Field/FieldOptions/calculator/index.vue]: Please remove this file and just change when touch it

File: src/components/ADempiere/Field/FieldOptions/translated/index.vue [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <!--
  ADempiere-Vue (Frontend) for ADempiere ERP & CRM Smart Business Solution
  Copyright (C) 2017-Present E.R.P. Consultores y Asociados, C.A.
- Contributor(s): Edwin Betancourt edwinBetanc0urt@hotmail.com www.erpya.com
```
Comment (yamelsenih) [inline src/components/ADempiere/Field/FieldOptions/translated/index.vue]: Please remove this file and just change when touch it

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: src/components/ADempiere/Field/FieldOptions/calculator/index.vue (hunk 5 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/Field/FieldOptions/calculator/index.vue]: changes applied

File: src/components/ADempiere/Field/FieldOptions/translated/index.vue (hunk 6 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/Field/FieldOptions/translated/index.vue]: changes applied

File: src/components/ADempiere/Field/FieldOptions/calculator/buttons.js (hunk 4 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/Field/FieldOptions/calculator/buttons.js]: changes applied

**Comment by EdwinBetanc0urt** [comment]
@oscarinak  for text fields by default the `like` (contains) operator is used, and this operator as well as the `not like` (not contains) operator must be used with the %% symbols to find matches. 

For example `%Aire%` will find records that contain the word `Aire`, and `Aire%` will find records that begin with the word `Aire`, conversely `%Aire` will find records that end with the word `Aire` as in the case of the product `Tope Electrico 30 Pulg. 4 Hornillas Negro Ffec3025ub Frigidaire`.

https://user-images.githubusercontent.com/20288327/120015629-8b2c8780-bfb1-11eb-865a-b6438d2a0c0e.mp4

**Pull request merged**